iib/web/models.py

```diff
@@ -1943,3 +1949,133 @@ def get_mutable_keys(self) -> Set[str]:
         rv = super().get_mutable_keys()
         rv.add('parent_bundle_image_resolved')
         return rv
+
+
+class RequestFbcOperations(Request, RequestIndexImageMixin):
+    """FBC operation build request."""
+
+    __tablename__ = 'request_fbc_operations'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+
+    fbc_fragment_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    fbc_fragment_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    fbc_fragment = db.relationship('Image', foreign_keys=[fbc_fragment_id], uselist=False)
+    fbc_fragment_resolved = db.relationship(
+        'Image', foreign_keys=[fbc_fragment_resolved_id], uselist=False
+    )
+    build_tags = None
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['fbc_operations'].value
+    }
+
+    @classmethod
+    def from_json(  # type: ignore[override] # noqa: F821
+        cls,
+        kwargs: FbcOperationRequestPayload,
+    ):
+        """
+        Handle JSON requests for the fbc-operations API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        validate_request_params(
+            request_kwargs,
+            required_params={'fbc_fragment', 'from_index'},
+            optional_params={
+                'add_arches',
+                'overwrite_from_index_token',
+                'overwrite_from_index',
+                'binary_image',
+                'build_tags',
+            },
+        )
+
+        # Validate parent_bundle_image is correctly provided
+        fbc_fragment = request_kwargs.get('fbc_fragment')
+        if not isinstance(fbc_fragment, str):
+            raise ValidationError('The "fbc_fragment" must be a string')
+        request_kwargs['fbc_fragment'] = Image.get_or_create(pull_specification=fbc_fragment)
+
+        from_index = request_kwargs.get('from_index', None)
```

Comment:
Can I ask why aren't we using `cls.from_json` for things like `from_index`, `binary_image`, `ovewrite_from_index` etc. ?

Similar to the call in RequestAdd `from_json` function. Line 1087

Reply:
I am sorry but I see that in `RequestAdd` it is done same way as I did: `using request_kwargs.get()`
What am I missing?

Reply:
Since you are inheriting `RequestIndexImageMixin`, you can probably reuse the `from_json` function to verify `index_image, binary_image, etc`. 

I meant you can call the parent class from_json function like https://github.com/release-engineering/iib/blob/master/iib/web/models.py#L1087 instead of re-writing these branches again. Am I missing something?